tests/ha/ocfs2.pm

```diff
@@ -1,36 +1,90 @@
-# SUSE's openQA tests
-#
-# Copyright © 2009-2013 Bernhard M. Wiedemann
-# Copyright © 2012-2016 SUSE LLC
-#
-# Copying and distribution of this file, with or without modification,
-# are permitted in any medium without royalty provided the copyright
-# notice and this notice are preserved.  This file is offered as-is,
-# without any warranty.
-
-use base "installbasetest";
+use base "hacluster";
 use testapi;
 use autotest;
+use lockapi;
 
 sub run() {
-    assert_and_click 'hawk-wizard-icon';
-    assert_and_click 'hawk-ocfs2-wizard';
-    assert_and_click 'hawk-wizard-next';
-    assert_and_click 'hawk-block-device';
-    type_string '/dev/path/to/storage/device';
-    assert_and_click 'hawk-wizard-next';
-    assert_screen 'hawk-wizard-confirm';
-    assert_and_click 'hawk-wizard-next';
-    assert_screen 'hawk-dashboard';
-    send_key 'alt-f4';
-    type_string "ssh 10.0.2.16 -l root\n";
-    sleep 1;
-    type_string "openqaha\n";
-    sleep 1;
-    type_string "mkfs.ocfs2 /dev/path/to/storage/device\n";
-    type_string "crm resource online clusterfs\n";
-    type_string "crm status\n";
-    assert_screen 'crm-ocfs-running';
+    my $ocfs2_partition = "/dev/disk/by-path/ip-*-lun-2";
+    # M1 belongs to node2, M2 belongs to node1
+    if (!get_var("HACLUSTERJOIN")) {    #node1
+        mutex_unlock("MUTEX_HA_" . get_var("CLUSTERNAME") . "_M2");
+        type_string "echo wait until DLM resource is created\n";
+        sleep 10;
+        mutex_lock("MUTEX_HA_" . get_var("CLUSTERNAME") . "_M1");
+    }
+    else {                              #node2
+        type_string qq(EDITOR="sed -ie '\$ a primitive dlm ocf:pacemaker:controld op monitor interval=60 timeout=60'" crm configure edit; echo dlm_add=\$? > /dev/$serialdev\n);
+        die "create DLM resource failed" unless wait_serial "dlm_add=0", 60;
+        type_string qq(EDITOR="sed -ie '\$ a group base-group dlm'" crm configure edit; echo base_group_add=\$? > /dev/$serialdev\n);
+        die "create base-group failed" unless wait_serial "base_group_add=0", 60;
+        type_string qq(EDITOR="sed -ie '\$ a clone base-clone base-group'" crm configure edit; echo base_clone_add=\$? > /dev/$serialdev\n);
+        die "create base-clone failed" unless wait_serial "base_clone_add=0", 60;
+        sleep 5;                        # 5 seconds should be enough to start dlm_controld on all nodes
```

Comment:
replace with what? any idea how to synchronise threads? mutexes are usually used to control access to the shared variables, we don't have shared anything, only mutexes, and I don't have better ideas.